### DISCUSSION TITLE
Grouping by date using Mongo's $dateToString feature

### DISCUSSION THREAD

**Discussion by viveleroi** [prisma/prisma#16930] [Q&A] [answered]
I'm trying to port a MongoDB/Mongoose query to Prisma.

I have an aggregation query that groups record by date - but *only* the year/month/day and discards the time.

In Mongo, this was done using `$dateToString`:

```js
$group: {
  _id: { $dateToString: { format: '%Y-%m-%d', date: '$date' } },
  count: { $sum: 1 }
}
```

I can't seem to find a way to do this in prisma.

Here is my full mongodb query:

```js
const promotions = await Promotion.aggregate([{
  $match: {
    date: {
      $gte: start,
      $lt: end
    }
  }
}, {
  $group: {
    _id: { $dateToString: { format: '%Y-%m-%d', date: '$date' } },
    count: { $sum: 1 }
  }
}, {
  $sort: { _id: 1 }
}])
```

EDIT: I've found Prisma's "raw" functions, so I think my only choice here is to use  `aggregateRaw`

**Comment by nurul3101** [comment, marked as answer]
Hello @viveleroi 👋 

I am glad to hear that you were able to find [aggregateRaw](https://www.prisma.io/docs/concepts/components/prisma-client/raw-database-access#aggregateraw).
That would be the only option for implementing this query.

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
